SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/VisibilityTest/VisibilityTest.cpp

```diff
@@ -0,0 +1,336 @@
+		}, 5.0f);
+	}
+
+	{	// Step 7 - Server Set Actor Hidden.
+		AddStep(TEXT("ServerSetActorHidden"), FWorkerDefinition::Server(1), nullptr, [this](ASpatialFunctionalTest* NetTest) {
+			int Counter = 0;
+			int ExpectedReplicatedActors = 1;
+			for (TActorIterator<AReplicatedVisibilityTestActor> Iter(GetWorld()); Iter; ++Iter)
+			{
+				Counter++;
+				Iter->SetHidden(true);
+				AssertEqual_Bool(Iter->IsHidden(), true, TEXT("TestHiddenActors is Hidden in the server world"), NetTest);
+			}
+			AssertEqual_Int(Counter, ExpectedReplicatedActors, TEXT("Number of TestHiddenActors in server world"), NetTest);
+			FinishStep();
+		});
+	}
+
+	{	// Step 8 - Observe if the test actor has been replicated to the server.
+		AddStep(TEXT("ServerCheckReplicatedActorsAfterSetActorHidden"), FWorkerDefinition::Server(1), nullptr, [this](ASpatialFunctionalTest* NetTest)
+		{
+			int Counter = 0;
+			int ExpectedReplicatedActors = 1;
+			for (TActorIterator<AReplicatedVisibilityTestActor> Iter(GetWorld()); Iter; ++Iter)
+			{
+					Counter++;
+			}
+			AssertEqual_Int(Counter, ExpectedReplicatedActors, TEXT("Number of TestHiddenActors in the server world"), NetTest);
+			FinishStep();
+		});
+	}
+
+	{	// Step 9 - Observe if the test actor has been replicated to the Clients.
+		AddStep(TEXT("ClientCheckReplicatedActorsAfterSetActorHidden"), FWorkerDefinition::AllClients, nullptr, [this](ASpatialFunctionalTest* NetTest)
+		{
+			int Counter = 0;
+			int ExpectedReplicatedActors = 0;
+			for (TActorIterator<AReplicatedVisibilityTestActor> Iter(GetWorld()); Iter; ++Iter)
+			{
+				Counter++;
+			}
+			AssertEqual_Int(Counter, ExpectedReplicatedActors, TEXT("Number of TestHiddenActors in the client world"), NetTest);
+			FinishStep();
+		});
+	}
+
+	{	// Step 10 - Server moves Client 1 close to the cube.
+		AddStep(TEXT("ServerMoveClient1CloseToCube"), FWorkerDefinition::Server(1), nullptr, [this](ASpatialFunctionalTest* NetTest)
+		{
+			ASpatialFunctionalTestFlowController* FlowController = GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1);
+			checkf(FlowController, TEXT("Can't be running test without valid FlowControl."));
+			APlayerController* PlayerController = Cast<APlayerController>(FlowController->GetOwner());
+			ATestMovementCharacter* PlayerCharacter = Cast<ATestMovementCharacter>(PlayerController->GetPawn());
+			if (PlayerCharacter->GetActorLocation().Equals(CharacterRemoteLocation, 50.0f))
+			{
+				if (PlayerCharacter->SetActorLocation(Character1StartingLocation))
+				{
+					if (PlayerCharacter->GetActorLocation().Equals(Character1StartingLocation, 50.0f))
+					{
+						FinishStep();
+					}
+				}
+			}
+		});
+	}
+
+	{   // Step 11 - Make sure that the Client moved to the correct location.
+		AddStep(TEXT("ClientCheckSecondMovement"), FWorkerDefinition::Client(1), nullptr, nullptr, [this](ASpatialFunctionalTest* NetTest, float DeltaTime)
+		{
+			ASpatialFunctionalTestFlowController* FlowController = GetLocalFlowController();
```

Comment:
Feel like we could refactor this check actor location flow for a controllers character into a helper function. I'm not insistent on it though, just seems like a lot of lines could be saved.